WordPress/Classes/ViewRelated/Reader/DetailWebView/ReaderDetailToolbar.swift

```diff
@@ -0,0 +1,323 @@
+
+        likeCountObserver = post.observe(\.likeCount, options: .new) { [weak self] _, _ in
+            self?.configureLikeActionButton(true)
+        }
+
+        commentCountObserver = post.observe(\.commentCount, options: .new) { [weak self] _, _ in
+            self?.configureCommentActionButton()
+        }
+
+        configureActionButtons()
+    }
+
+    deinit {
+        likeCountObserver?.invalidate()
+        commentCountObserver?.invalidate()
+    }
+
+    // MARK: - Actions
+
+    @IBAction func didTapSaveForLater(_ sender: Any) {
+        guard let readerPost = post, let context = readerPost.managedObjectContext,
+            let viewController = viewController as? UIViewController & UIViewControllerTransitioningDelegate else {
+            return
+        }
+
+        if !readerPost.isSavedForLater {
+            FancyAlertViewController.presentReaderSavedPostsAlertControllerIfNecessary(from: viewController)
+        }
+
+        ReaderSaveForLaterAction().execute(with: readerPost, context: context, origin: .postDetail) { [weak self] in
+            self?.saveForLaterButton.isSelected = readerPost.isSavedForLater
+            self?.prepareActionButtonsForVoiceOver()
+        }
+    }
+
+    @IBAction func didTapReblog(_ sender: Any) {
+        guard let post = post, let viewController = viewController else {
+            return
+        }
+
+        ReaderReblogAction().execute(readerPost: post, origin: viewController, reblogSource: .detail)
+    }
+
+    @IBAction func didTapComment(_ sender: Any) {
+        guard let post = self.post, let viewController = viewController else {
+            return
+        }
+
+        ReaderCommentAction().execute(post: post, origin: viewController)
+    }
+
+    @IBAction func didTapLike(_ sender: Any) {
+        guard let post = post else {
+            return
+        }
+
+        if !post.isLiked {
+            UINotificationFeedbackGenerator().notificationOccurred(.success)
+        }
+
+        let service = ReaderPostService(managedObjectContext: post.managedObjectContext!)
+        service.toggleLiked(for: post, success: nil, failure: { [weak self] (error: Error?) in
+            self?.trackArticleDetailsLikedOrUnliked()
+            if let anError = error {
+                DDLogError("Error (un)liking post: \(anError.localizedDescription)")
+            }
+        })
+    }
+
+    // MARK: - Styles
+
+    private func applyStyles() {
+        backgroundColor = .listForeground
+        dividerView.backgroundColor = .divider
+
+        WPStyleGuide.applyReaderCardActionButtonStyle(commentButton)
+        WPStyleGuide.applyReaderCardActionButtonStyle(likeButton)
+    }
+
+    // MARK: - Configuration
+
+    private func configureActionButtons() {
+        resetActionButton(likeButton)
+        resetActionButton(commentButton)
+        resetActionButton(saveForLaterButton)
+        resetActionButton(reblogButton)
+
+        guard let post = post else {
+            assertionFailure()
```

Comment:
why do we have assertion failure here but just `return` in other guards for post ?

Reply:
Probably different people writing code?

To keep consistency I removed this in 9c1775a272640aa992fe4b7a4cabbdccc7a3dcba